docs/explainer.md

```diff
@@ -9,6 +9,96 @@ See [Discourse](https://discourse.wicg.io/t/filling-the-remaining-gap-between-we
 User agents will need to carefully consider when to make the Raw Sockets API available to web applications,
 and what UI will be shown to the user.
 
+
+
+### Threat
+
+MITM attackers may inject sockets API calls into a web page.
+
+### Mitigation
+
+The API will only available in secure contexts (HTTPS).
+
+
+
+### Threat
+
+A web app might initiate connections without the user realizing.
+
+### Mitigation
+
+When initiating a connection, the user agent would show some UI.
+
+The user can be asked to specify a hostname or IP address, with an option to permit future connections to this host.
+
+
+
+### Threat
+
+Attackers may use the API to DDOS third parties.
+
+### Mitigation
+
+Connection attempts would be rate limited.
```

Comment:
This still allows for passive scans. Maybe this would be in control of the connection UI... the user could then hit a "retry" button, rather than the web page doing it.

Reply:
The UI would have a control like "[ ] Allow future connections", both to support resilience when network difficulties occur and to avoid permissions fatigue when a given web app is used frequently.

In an industrial context, a dedicated machine might be configured with a web app that connects to specific devices. Requiring user interaction after each device power cycle or other disconnection would be unfortunate.